frontend/src/components/common/Table.tsx

```diff
@@ -0,0 +1,244 @@
+import { Paper } from '@mui/material';
+import { useTheme } from '@mui/styles';
+import {
+  MaterialReactTable,
+  MRT_ColumnDef as MaterialTableColumn,
+  MRT_TableOptions as MaterialTableOptions,
+  useMaterialReactTable,
+} from 'material-react-table';
+import { useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import helpers from '../../helpers';
+import { useURLState } from '../../lib/util';
+import { useSettings } from '../App/Settings/hook';
+import Empty from './EmptyContent';
+import Loader from './Loader';
+
+/**
+ * @see https://www.material-react-table.com/docs/api/column-options
+ */
+export type TableColumn<RowItem extends Record<string, any>, Value = any> = MaterialTableColumn<
+  RowItem,
+  Value
+> & { gridTemplate?: string | number };
+
+/**
+ * All the options provided by the MRT and some of our custom behaviour
+ *
+ * @see https://www.material-react-table.com/docs/api/table-options
+ */
+export type TableProps<RowItem extends Record<string, any>> = Omit<
+  MaterialTableOptions<RowItem>,
+  'columns'
+> & {
+  columns: TableColumn<RowItem>[];
+  emptyMessage?: string;
+  errorMessage?: any;
+  /** Whether to reflect the page/perPage properties in the URL.
+   * If assigned to a string, it will be the prefix for the page/perPage parameters.
+   * If true or '', it'll reflect the parameters without a prefix.
+   * By default, no parameters are reflected in the URL. */
+  reflectInURL?: string | boolean;
+  initialPage?: number;
```

Comment:
Can you please add documentation for these fields? It helps on hover, and in the storybook to understand what they are (for).